targets/targets.json

```diff
@@ -630,6 +630,19 @@
         "release_versions": ["2", "5"],
         "device_name": "MK82FN256xxx15"
     },
+    "LPC54114": {
```

Comment:
I just talked to @maclobdell Offline, and he clarified the mbed database thing. @0xc0170 I think there is a (pretty arbitrary) restriction on target name length in the online compiler. `LPC54114` is a pretty reasonable shortening of `LPCXpresso54114`.